core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -0,0 +1,287 @@
+var createCollectionButtonSelector = '.protractor-test-create-collection';
+
+var createTopicButtonSelector = '.protractor-test-create-topic-button';
+var topicNameField = '.protractor-test-new-topic-name-field';
+var topicUrlFragmentField = '.protractor-test-new-topic-url-fragment-field';
+var topicDescriptionField = '.protractor-test-new-topic-description-field';
+var topicThumbnailButton = '.protractor-test-photo-button';
+var topicUploadButton = '.protractor-test-photo-upload-input';
+var topicPhotoSubmit = '.protractor-test-photo-upload-submit';
+var thumbnailContainer = '.protractor-test-thumbnail-container';
+var confirmTopicCreationButton =
+  '.protractor-test-confirm-topic-creation-button';
+var createdTopicLink = '.protractor-test-topic-name';
+
+var createStoryButtonSelector = '.protractor-test-create-story-button';
+var storyNameField = '.protractor-test-new-story-title-field';
+var storyUrlFragmentField = '.protractor-test-new-story-url-fragment-field';
+var storyDescriptionField = '.protractor-test-new-story-description-field';
+var storyThumbnailButton = '.protractor-test-photo-button';
+var storyUploadButton = '.protractor-test-photo-upload-input';
+var storyPhotoSubmit = '.protractor-test-photo-upload-submit';
+var thumbnailContainer = '.protractor-test-thumbnail-container';
+var confirmStoryCreationButton =
+  '.protractor-test-confirm-story-creation-button';
+
+var createSkillButtonSelector = '.puppeteer-test-add-skill-button';
+var skillDescriptionField = '.protractor-test-new-skill-description-field';
+var skillOpenConceptCard = '.protractor-test-open-concept-card';
+var skillEditor = '.protractor-test-concept-card-text';
+var confirmSkillCreationButton =
+  '.protractor-test-confirm-skill-creation-button';
+var skillReviewMaterialInput = '.oppia-rte';
+
+const login = async function(browser, page) {
+  try {
+    // eslint-disable-next-line dot-notation
+    await page.goto(
+      ADMIN_URL, { waitUntil: 'networkidle0' });
+    await page.waitForSelector('#admin', {visible: true});
+    await page.click('#admin');
+    await page.click('#submit-login');
+    // This is to check if the user's account was already made.
+    try {
+      await page.waitForSelector('#username', {visible: true});
+      await page.type('#username', 'username1');
+      await page.click('#terms-checkbox');
+      await page.waitForSelector('#signup-submit');
+      await page.click('#signup-submit');
+      await page.waitForSelector('.oppia-navbar-dropdown-toggle');
+    } catch (error) {
+      // Already Signed in.
+    }
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.log('Login Failed');
+    // eslint-disable-next-line no-console
+    console.log(e);
+  }
+};
+
+const setRole = async function(browser, page, role) {
+  try {
+    // eslint-disable-next-line dot-notation
+    await page.goto(
+      'http://127.0.0.1:8181/admin#/roles', { waitUntil: 'networkidle0' });
```

Comment:
What is `networkidle0`? Can you create a variable with a descriptive name for it (also allows you to re-use the the variable below)?

Reply:
Gotcha. Can you still create a variable for the string though? And maybe add a comment with a link to the above.

Reply:
Done